merge-by/scripts/bump-dates.js

```diff
@@ -0,0 +1,48 @@
+module.exports = async ({ github, context }) => {
+    // Ignore PR "opened" events if the PR was opened as draft
+    const wasJustOpened = context.action === "opened";
+    if (wasJustOpened && context.payload.pull_request.draft) {
+        return;
+    }
+
+    const wasJustPushed = context.action === "synchronize";
+    
+    const owner = context.repo.owner;
+    const repo = context.repo.repo;
+    const comments = (await github.rest.issues.listComments({ owner, repo, issue_number: context.payload.pull_request.number }))?.data;
+    const existingComment = comments?.find((comment) => 
+        comment.user.login === "github-actions[bot]" && comment.body.includes("**📅 Suggested merge-by date:"));
+    
+    // For existing PRs, only post the date if the PR isn't in draft.
+    if (context.payload.pull_request.draft) {
+        return;
+    }
+    
+    // Determine new merge-by date based on the last time the PR was marked as ready
+    const currentTime = new Date();
+    const mergeBy = new Date();
+    mergeBy.setDate(currentTime.getDate() + 14);
```

Comment:
In the future we may want to consider making the 14 days value a customizable input that can be passed in to the action 😋